backend/test_server.py

```
@@ -189,15 +189,297 @@ def make_post_request(endpoint, data):
+    result = response.json()
+    assert 'error' in result
+    print(f"✓ MCQ answer with mismatched lists correctly rejected: {result['error']}")
+
+
+def test_get_content_invalid_url():
+    """Test rejection of invalid URL format."""
+    endpoint = '/get_content'
+    data = {
+        'url': 'not-a-valid-url'
+    }
+    response = requests.post(f'{BASE_URL}{endpoint}', headers={'Content-Type': 'application/json'}, data=json.dumps(data))
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)